File: knowledge_base_service/tests/fixtures/code_samples/sample.py
"""Sample Python module for testing tree-sitter parser."""

import os
from typing import List, Optional
from dataclasses import dataclass


@dataclass
class Person:
    """A person class."""

    name: str
    age: int

    def greet(self) -> str:
        """Return a greeting message."""
        return f"Hello, my name is {self.name}"

    def celebrate_birthday(self) -> None:
        """Increment age by 1."""
        self.age += 1


class Calculator:
    """A simple calculator class."""

    def __init__(self) -> None:
        self.history: List[float] = []

    def add(self, a: float, b: float) -> float:
        """Add two numbers."""
        result = a + b
        self.history.append(result)
        return result

    def subtract(self, a: float, b: float) -> float:
        """Subtract b from a."""
        result = a - b
        self.history.append(result)
        return result

    def get_history(self) -> List[float]:
        """Get calculation history."""
        return self.history.copy()


def standalone_function(x: int) -> int:
    """A standalone function."""
    return x * 2


def process_data(data: Optional[List[str]]) -> List[str]:
    """Process data with optional input."""
    if data is None:
        return []
    return [item.upper() for item in data]

File: knowledge_base_service/tests/fixtures/code_samples/sample.rs
//! Sample Rust module for testing tree-sitter parser

use std::collections::HashMap;

/// A person struct
pub struct Person {
    name: String,
    age: u32,
}

impl Person {
    /// Create a new Person
    pub fn new(name: &str, age: u32) -> Self {
        Self {
            name: name.to_string(),
            age,
        }
    }

    /// Get a greeting message
    pub fn greet(&self) -> String {
        format!("Hello, my name is {}", self.name)
    }

    /// Celebrate birthday
    pub fn celebrate_birthday(&mut self) {
        self.age += 1;
    }

    /// Get name
    pub fn get_name(&self) -> &str {
        &self.name
    }

    /// Get age
    pub fn get_age(&self) -> u32 {
        self.age
    }
}

/// Calculator struct
pub struct Calculator {
    history: Vec<f64>,
}

impl Calculator {
    /// Create a new Calculator
    pub fn new() -> Self {
        Self {
            history: Vec::new(),
        }
    }

    /// Add two numbers
    pub fn add(&mut self, a: f64, b: f64) -> f64 {
        let result = a + b;
        self.history.push(result);
        result
    }

    /// Subtract b from a
    pub fn subtract(&mut self, a: f64, b: f64) -> f64 {
        let result = a - b;
        self.history.push(result);
        result
    }

    /// Get calculation history
    pub fn get_history(&self) -> &[f64] {
        &self.history
    }
}

impl Default for Calculator {
    fn default() -> Self {
        Self::new()
    }
}

/// Standalone function
pub fn process_data(data: &[String]) -> Vec<String> {
    data.iter()
        .map(|s| s.to_uppercase())
        .collect()
}

/// Generic function
pub fn identity<T>(value: T) -> T {
    value
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_person() {
        let person = Person::new("Alice", 30);
        assert_eq!(person.get_age(), 30);
    }
}
